move define clients/luns to lib

mikechristie opened the pull request:
Move the define clients/luns code to ceph_iscsi_config lib
so cli can use it.

Questions: The added lib functions return true/false which did not seem very pythony. Should they be throwing exceptions? I did not see any ceph-iscsi-config defined ones.

dillaman commented:
@mikechristie I'd vote for throwing exceptions within these helpers and catching where appropriate

Commit:
move define clients/luns to lib v2:

Move the define clients/luns code to ceph_iscsi_config lib
so cli can use it.

Changes:
v2 - Add CephiSCSIError exception and return instead of True/False

Signed-off-by: Mike Christie <mchristi@redhat.com>

mikechristie commented:
Updated to throw exceptions.

dillaman reviewed (approved):
lgtm
